pkgs/tools/misc/redtimer/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, ruby
+, bundlerEnv
+, livesplit-core
+, ncurses
+, toilet
+}:
+
+let
+  gems = bundlerEnv {
+    name = "redtimer-env";
+    gemdir = ./.;
+    inherit ruby;
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "redtimer";
+  version = "unstable-2021-08-01";
+
+  src = fetchFromGitHub {
+    owner = "cout";
+    repo = pname;
+    rev = "ab5ac1b527d670bc390456d68f3520cd8bd8e1cc";
+    sha256 = "1ghlqir088azr65nzbaw2i2g4k417rfx5n8h2c3pywysv2vfxk0y";
+  };
+
+  preConfigure = ''
+    ln -s ${livesplit-core}/lib/{liblivesplit_core.so,bindings/LiveSplitCore.rb} .
+  '';
+
+  nativeBuildInputs = [ makeWrapper ruby gems toilet ];
+  buildInputs = [ ncurses ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,share}
```

Comment:
I do think this is a much nicer solution, as to avoid unnecessary code repetition. If you don't feel strongly about it, I'd like to keep it like as is.